lib/Doctrine/ORM/Query/Exec/AbstractSqlExecutor.php

```diff
@@ -60,4 +72,15 @@ public function removeQueryCacheProfile()
      * @return Result|int
      */
     abstract public function execute(Connection $conn, array $params, array $types);
+
+    /** @return list<string> */
+    public function __sleep(): array
+    {
+        return array_diff_key(array_keys((array) $this), ['*_sqlStatements']);
```

Comment:
This does not seem to work for old versions of PHP… or does it have to do with `VarExporter`? I still didn't quite get how it comes into play for https://github.com/doctrine/orm/blob/609647a51a1b28a674de270204a5cea3af57d554/lib/Doctrine/ORM/Query/ParserResult.php#L143-L147

Reply:
:thinking: why did I use `array_diff_key` here… makes no sense…